tools/java/org/unicode/cldr/util/CLDRLocale.java

```diff
@@ -368,13 +368,15 @@ public ULocale toULocale() {
      * @return
      */
     public static CLDRLocale getInstance(String s) {
-        if (s == null) return null;
-        CLDRLocale loc = stringToLoc.get(s);
-        if (loc == null) {
-            loc = new CLDRLocale(s);
-            loc.register();
+        synchronized(CLDRLocale.class) {
```

Comment:
If possible I would suggest using ConcurrentMap with computeIfAbsent():
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentMap.html#computeIfAbsent-K-java.util.function.Function-

Reply:
@hagbard that sounds really promising. Let's see if this PR will unblock and we can refine.